konan_sdk/konan_service/models.py

```diff
@@ -0,0 +1,34 @@
+import abc
+
+from konan_sdk.konan_service.serializers import (
+    KonanServiceBasePredictionRequest, KonanServiceBasePredictionResponse,
+    KonanServiceBaseEvaluateRequest, KonanServiceEvaluateResponse,
+)
+
+
+class KonanServiceBaseModel(abc.ABC):
+    @abc.abstractmethod
+    def predict(self, req: KonanServiceBasePredictionRequest) -> KonanServiceBasePredictionResponse:
+        """Predicts using the preprocessed_input
+
+        Args:
+            req (PredictionRequestClass): raw request data from API
+
+        Returns:
+            PredictionResponseClass: prediction.
+            This will be the response returned by the API.
+        """
+        pass
+
+    @abc.abstractmethod
+    def evaluate(self, req: KonanServiceBaseEvaluateRequest) -> KonanServiceEvaluateResponse:
+        """Evaluates the model using past predictions and their feedback
+
+        Args:
+            req (KonanServiceBaseEvaluateRequest): raw request data from API
+
+        Returns:
+            KonanServiceBaseEvaluateResponse: evaluation.
```

Comment:
Minor typo there with `KonanServiceBaseEvaluateResponse` in docstring, though I think it fits the naming scheme better if it does include `Base` like the other classes

Reply:
Will fix it.